src/arrayMethodJoin.js

```diff
@@ -1,11 +1,36 @@
+/* eslint-disable use-isnan */
+/* eslint-disable no-console */
 'use strict';
 
 /**
  * Implement method join
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    const sep = String(separator);
+
+    for (let i = 0; i < this.length; i++) {
+      switch (true) {
+        case (i + 1) !== this.length && this[i] === undefined:
```

Comment:
no need to check if it's the last element every time

instead of doing it, you may just add `this[i]` or an empty string if it's `null` or `undefined`.
and then in another if statement add a separator if it's not the last element